### DISCUSSION TITLE
Gnome / dconf

### DISCUSSION THREAD

**Discussion by tandy-1000** [twpayne/chezmoi#2101] [Q&A] [answered]
Hi, I'm trying to use chezmoi to manage my dconf, but I am not aware of how to make sure that changes I make to my dconf outside of chezmoi (e.g. in Gnome settings) are dumped so that I can store them in my git repo.

I found this [guide](https://peterbabic.dev/blog/keep-gnome-shell-settings-dotfiles-yadm/) to do it using hooks with yadm, but I'm not sure how to do this with chezmoi.

Thank you.

**Comment by nandalopes** [comment]
I used to have configuration for Tilix Terminal, which uses dconf. I kept two files:

- `current-Tilix`, a `modify_` file with dumped config
- `tracked-Tilix`, with my default, initial commit

I could also have a `run_onchange_` script loading `tracked-Tilix` whenever it changes.

**Reply by tandy-1000** [reply]
I see, this is a per application config, I am thinking it will be easier to just store my whole dconf using dconf dump.

**Comment by twpayne** [comment, marked as answer]
Supporting dconf is [an example in the user guide](https://www.chezmoi.io/user-guide/use-scripts-to-perform-actions/#run-a-script-when-the-contents-of-another-file-changes).

**Reply by tandy-1000** [reply]
Yes, I have used this example to help when loading my dconf file when applying, but I am looking for a solution to dump my dconf file like in the yadm guide that I linked.

Maybe my best option is to just use pre-commit hooks rather than some chezmoi feature?

**Reply by twpayne** [reply]
Given that dumping your dconf file requires running:

    dconf dump / > $(chezmoi source-path)/dconf.ini

There are many ways to achieve this, including:
* Using a git pre-commit hook is possible, but will pollute your git commits with updates to this file, and may result in saving an undesired dconf.ini if the hook runs while you have made temporary changes to your settings or changes that are specific to the current machine.
* You can use an alias, shell function, or shell script to run this manually when you want.

I would recommend running the command manually.

Alternatively, my approach is to set the settings I care about with a `run_onchange_after_` script. [Here's mine](https://github.com/twpayne/dotfiles/blob/master/home/.chezmoiscripts/linux/run_onchange_after_configure-gnome.sh.tmpl).

**Reply by tandy-1000** [reply]
I see, that makes sense, I will just run it manually once and add any settings I care about manually later.
